symfony/client/src/orangehrmTimePlugin/components/AddCustomerModal.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <oxd-dialog
+    @update:show="onCancel"
+    :style="{width: '90%', maxWidth: '450px'}"
+  >
+    <div class="orangehrm-modal-header">
+      <oxd-text type="card-title">Add Customer</oxd-text>
+    </div>
+    <oxd-divider />
+    <oxd-form :loading="isLoading" @submitValid="onSave">
+      <oxd-form-row>
+        <oxd-input-field
+          :label="$t('general.name')"
+          v-model="customer.name"
+          :rules="rules.name"
+          required
+        />
+      </oxd-form-row>
+      <oxd-form-row>
+        <oxd-input-field
+          type="textarea"
+          label="Description"
+          placeholder="Type description here"
+          v-model="customer.description"
+          :rules="rules.description"
+        />
+      </oxd-form-row>
+      <oxd-divider />
+      <oxd-form-actions>
+        <required-text />
+        <oxd-button
+          displayType="ghost"
+          :label="$t('general.cancel')"
+          @click="onCancel"
+        />
+        <submit-button />
+      </oxd-form-actions>
+    </oxd-form>
+  </oxd-dialog>
+</template>
+
+<script>
+import {APIService} from '@ohrm/core/util/services/api.service';
+import {
+  required,
+  shouldNotExceedCharLength,
+} from '@ohrm/core/util/validation/rules';
+import Dialog from '@ohrm/oxd/core/components/Dialog/Dialog';
+import promiseDebounce from '@ohrm/oxd/utils/promiseDebounce';
+
+const customerModel = {
+  id: '',
+  name: '',
+  description: '',
+};
+
+export default {
+  name: 'add-customer-modal',
+  components: {
+    'oxd-dialog': Dialog,
+  },
+  data() {
+    return {
+      isLoading: false,
+      customer: {...customerModel},
+      rules: {
+        name: [
+          required,
+          shouldNotExceedCharLength(50),
+          promiseDebounce(this.validateCustomerName, 500),
+        ],
+        description: [shouldNotExceedCharLength(250)],
+      },
+    };
+  },
+  setup() {
+    const http = new APIService(
+      window.appGlobal.baseUrl,
+      '/api/v2/time/customers',
+    );
+    return {
+      http,
+    };
+  },
+  methods: {
+    onSave() {
+      this.isLoading = true;
+      this.http
+        .create({
+          name: this.customer.name,
+          description: this.customer.description,
+        })
+        .then(() => {
+          return this.$toast.saveSuccess();
+        })
+        .then(() => {
+          this.onCancel();
+        });
+    },
+    onCancel() {
+      this.$emit('close');
```

Comment:
define [emits](https://v3.vuejs.org/api/options-data.html#emits). make sure to get latest and run yarn lint